app/controllers/candidate_interface/unsubmitted_application_form_controller.rb

```diff
@@ -46,6 +48,15 @@ def submit
 
   private
 
+    def track_adviser_offering
+      event = DfE::Analytics::Event.new
+        .with_type(:candidate_offered_adviser)
+        .with_user(current_user)
+        .with_request_details(request)
+
+      DfE::Analytics::SendEvents.do([event])
+    end
+
```

Comment:
👍 I'll pull these out into a class; I'm not sure I see the benefit of constants over symbols here though as we'd end up with something like:

```
CANDIDATE_SIGNED_UP_FOR_ADVISER_EVENT = :candidate_signed_up_for_adviser

def candidate_signed_up_for_adviser
  DfE::Analytics::Event.new.with_type(CANDIDATE_SIGNED_UP_FOR_ADVISER_EVENT)
  ...
end

def '#candidate_signed_up_for_adviser'
  it 'enqueues a candidate_signed_up_for_adviser event' do
    expect(described_class::CANDIDATE_SIGNED_UP_FOR_ADVISER_EVENT).to have_been_enqueued_as_analytics_events
  end
end
```

If we ever changed the event name/symbol you'd likely still end up renaming the constant, method, spec etc to be consistent so I'm not sure what benefit the constant brings over a symbol? happy to discuss/be convinced otherwise though if you feel strongly about it?